tools/flex/patches/300-m4-path.patch

```diff
@@ -0,0 +1,23 @@
+--- a/src/main.c
++++ b/src/main.c
+@@ -213,6 +213,8 @@ int main (int argc, char *argv[])
+ 
+ void check_options (void)
+ {
++	const char * staging_dir = NULL;
++	char * m4_staging = NULL;
+ 	int     i;
+     const char * m4 = NULL;
+ 
+@@ -341,7 +343,10 @@ void check_options (void)
+ 
+     /* Setup the filter chain. */
+     output_chain = filter_create_int(NULL, filter_tee_header, headerfilename);
+-    if ( !(m4 = getenv("M4"))) {
++    if ( (staging_dir = getenv("STAGING_DIR_HOST"))) {
++	asprintf(&m4_staging, "%s/bin/m4", staging_dir);
```

Comment:
asprintf... isn't this a memory leak?

Reply:
If `flex` was a daemon then it would be a memory leak, yes.  However, `flex` allocates memory at many occasions using `calloc` and rarely frees any of it. The rule of thumb seems to be that only if the allocation happens inside a loop the memory is also freed. If it is a single allocation it all relies on the executable being a one-shot tool and everything being freed once the process terminates.